config/routes.rb

```diff
@@ -184,10 +184,11 @@
       resources :sites, path: "locations", on: :member, except: %i[destroy show]
 
       scope module: "providers" do
-        resources :allocations, only: [:index], on: :member, param: :training_provider_code do
+        resources :allocations, only: %i{index}, on: :member, param: :training_provider_code do
           member do
-            post :requests
-            get :requests
+            get "/new", action: :new
```

Comment:
🤔 why does `:new` not work in this case?

Reply:
I'm guessing because it part of the 7 rails restful routes on resources in some how interferes

Reply:
Does adding `new create show` on line 187 not work? I think the URLs look a bit funny here too. Have we carried the 2020 bit over from other pages?

Reply:
short answer: no

long answer: the url is `https://localhost:3000/organisations/B20/2020/allocations/2BO/new`. it's not using standard rails nested resources routing. which would be `https://localhost:3000/organisations/B20/2020/training_providers/2BO/allocations/2BO/new` this would have solved all these problems. I have no idea why non rails routing was used in the first place. I'm happy to rip this out and change it if I can get some design input to make sure they happy with doing this

Reply:
Regarding `2020` that has been carried over and seems like all the other parts in publisher follow the same convention/pattern